var/spack/repos/builtin/packages/meep/package.py

```diff
@@ -40,12 +42,18 @@ class Meep(AutotoolsPackage):
     depends_on('blas',        when='+blas')
     depends_on('lapack',      when='+lapack')
     depends_on('harminv',     when='+harminv')
-    depends_on('guile',       when='+guile')
-    depends_on('libctl',      when='+libctl')
+    depends_on('guile@:2',    when='@:1.4+guile')
+    depends_on('guile@2:',    when='@1.4:+guile')
+    depends_on('libctl@3.2',  when='@:1.3+libctl')
+    depends_on('libctl@4:',   when='+libctl')
     depends_on('mpi',         when='+mpi')
     depends_on('hdf5~mpi',    when='+hdf5~mpi')
     depends_on('hdf5+mpi',    when='+hdf5+mpi')
     depends_on('gsl',         when='+gsl')
+    depends_on('python',      when='+python')
+    depends_on('py-numpy',    when='+python')
+    depends_on('py-mpi4py',   when='+python+mpi')
+    depends_on('swig',        when='+python')
```

Comment:
I think this can be written in a neater way using the `when` context manager:
```suggestion
    with when('+python'):
        depends_on('python')
        depends_on('py-numpy')
        depends_on('swig')
        depends_on('py-mpi4py', when='+mpi')
```

Reply:
@tomgreen66